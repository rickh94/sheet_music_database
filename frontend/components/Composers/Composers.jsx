import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import Header from '../Header'
import Container from 'react-bulma-components/lib/components/container'
import Box from 'react-bulma-components/lib/components/box'
import Level from 'react-bulma-components/lib/components/level/level'

export class Composers extends Component {
  static propTypes = {
    composers: PropTypes.object.isRequired
  }

  render() {
    const composerList = this.props.composers.list
    return (
      <React.Fragment>
        <Header />
        <Container>
          <Box>
            {composerList.map(composer => (
              <ComposerItem key={composer.id} composer={composer} />
            ))}
          </Box>
        </Container>
      </React.Fragment>
    )
  }
}

export default Composers

export function ComposerItem(props) {
  let dates = ''
  const born = props.composer.born ? props.composer.born : ''
  const died = props.composer.died ? props.composer.died : ''
  if (props.composer.born || props.composer.died) {
    dates = `(${born}-${died})`
  }
  return (
    <Level>
      <Link to={`/composer/${props.composer.id}`} className="level-item level-left">
        {props.composer.name} {dates}
      </Link>
      <div className="level-right" />
    </Level>
  )
}

ComposerItem.propTypes = {
  composer: PropTypes.object.isRequired
}
