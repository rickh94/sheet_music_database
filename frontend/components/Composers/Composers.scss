.little-padding-bottom {
  padding-bottom: 0.2rem;
}
