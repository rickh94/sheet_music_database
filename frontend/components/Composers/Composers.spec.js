import React from 'react'
import { shallow } from 'enzyme'
import { Composers, ComposerItem } from './Composers'
import '../../setupTests'

describe('<Composers />', () => {
  const wrapper = shallow(<Composers composers={{ list: [] }} />)

  beforeEach(() => {
    wrapper.update()
  })

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the header', () => {
    expect(wrapper.exists('Connect(Header)')).toBeTruthy()
  })

  it('renders a container', () => {
    expect(wrapper.exists('Container')).toBeTruthy()
  })

  it('renders composers from props', () => {
    const composers = { list: [{ id: 1, name: 'Test' }, { id: 2, name: 'Test2' }] }
    wrapper.setProps({ composers })
    expect(wrapper.find('ComposerItem').length).toEqual(2)
  })
})

describe('ComposerItem', () => {
  const wrapper = shallow(<ComposerItem composer={{ name: '', era: '' }} />)
  const composer = { name: 'Johann Sebastian Bach', era: 'Baroque' }

  beforeEach(() => {
    wrapper.update()
  })

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a level', () => {
    expect(wrapper.exists('Level')).toBeTruthy()
  })

  it('renders a composer name with a link', () => {
    wrapper.setProps({ composer })
    expect(wrapper.contains(composer.name)).toBeTruthy()
    expect(wrapper.exists('Link')).toBeTruthy()
  })

  it('renders the dates if present', () => {
    wrapper.setProps({ composer: { ...composer, born: '1685', died: '1750' } })
    expect(wrapper.contains('(1685-1750)')).toBeTruthy()
  })

})
