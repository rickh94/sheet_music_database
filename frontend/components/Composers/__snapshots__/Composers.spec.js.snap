// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Composers /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Composers
    composers={
      Object {
        "list": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Container
          breakpoint={null}
          fluid={false}
          renderAs="div"
          style={Object {}}
        >
          <Box
            className=""
            renderAs="div"
            style={Object {}}
          />
        </Container>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": undefined,
          "breakpoint": null,
          "children": <Box
            className=""
            renderAs="div"
            style={Object {}}
          />,
          "className": undefined,
          "clearfix": undefined,
          "clipped": undefined,
          "fluid": false,
          "hidden": undefined,
          "invisible": undefined,
          "italic": undefined,
          "marginless": undefined,
          "overlay": undefined,
          "paddingless": undefined,
          "pull": undefined,
          "radiusless": undefined,
          "renderAs": "div",
          "responsive": undefined,
          "shadowless": undefined,
          "style": Object {},
          "textAlignment": undefined,
          "textColor": undefined,
          "textSize": undefined,
          "textTransform": undefined,
          "textWeight": undefined,
          "unselectable": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "children": Array [],
            "className": "",
            "clearfix": undefined,
            "clipped": undefined,
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Container
            breakpoint={null}
            fluid={false}
            renderAs="div"
            style={Object {}}
          >
            <Box
              className=""
              renderAs="div"
              style={Object {}}
            />
          </Container>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "breakpoint": null,
            "children": <Box
              className=""
              renderAs="div"
              style={Object {}}
            />,
            "className": undefined,
            "clearfix": undefined,
            "clipped": undefined,
            "fluid": false,
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": undefined,
              "children": Array [],
              "className": "",
              "clearfix": undefined,
              "clipped": undefined,
              "hidden": undefined,
              "invisible": undefined,
              "italic": undefined,
              "marginless": undefined,
              "overlay": undefined,
              "paddingless": undefined,
              "pull": undefined,
              "radiusless": undefined,
              "renderAs": "div",
              "responsive": undefined,
              "shadowless": undefined,
              "style": Object {},
              "textAlignment": undefined,
              "textColor": undefined,
              "textSize": undefined,
              "textTransform": undefined,
              "textWeight": undefined,
              "unselectable": undefined,
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ComposerItem renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComposerItem
    composer={
      Object {
        "era": "",
        "name": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backgroundColor": undefined,
      "breakpoint": null,
      "children": Array [
        <Link
          className="level-item level-left"
          replace={false}
          to="/composer/undefined"
        >
          
           
          
        </Link>,
        <div
          className="level-right"
        />,
      ],
      "className": "",
      "clearfix": undefined,
      "clipped": undefined,
      "hidden": undefined,
      "invisible": undefined,
      "italic": undefined,
      "marginless": undefined,
      "overlay": undefined,
      "paddingless": undefined,
      "pull": undefined,
      "radiusless": undefined,
      "renderAs": "div",
      "responsive": undefined,
      "shadowless": undefined,
      "style": Object {},
      "textAlignment": undefined,
      "textColor": undefined,
      "textSize": undefined,
      "textTransform": undefined,
      "textWeight": undefined,
      "unselectable": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "",
            " ",
            "",
          ],
          "className": "level-item level-left",
          "replace": false,
          "to": "/composer/undefined",
        },
        "ref": null,
        "rendered": Array [
          "",
          " ",
          "",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "level-right",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backgroundColor": undefined,
        "breakpoint": null,
        "children": Array [
          <Link
            className="level-item level-left"
            replace={false}
            to="/composer/undefined"
          >
            
             
            
          </Link>,
          <div
            className="level-right"
          />,
        ],
        "className": "",
        "clearfix": undefined,
        "clipped": undefined,
        "hidden": undefined,
        "invisible": undefined,
        "italic": undefined,
        "marginless": undefined,
        "overlay": undefined,
        "paddingless": undefined,
        "pull": undefined,
        "radiusless": undefined,
        "renderAs": "div",
        "responsive": undefined,
        "shadowless": undefined,
        "style": Object {},
        "textAlignment": undefined,
        "textColor": undefined,
        "textSize": undefined,
        "textTransform": undefined,
        "textWeight": undefined,
        "unselectable": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "",
              " ",
              "",
            ],
            "className": "level-item level-left",
            "replace": false,
            "to": "/composer/undefined",
          },
          "ref": null,
          "rendered": Array [
            "",
            " ",
            "",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "level-right",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
